
<!DOCTYPE html>
<html lang='en'>
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
    <title>Tree - rpms/libsmi - src.fedoraproject.org</title>
        <link rel="shortcut icon" type="image/vnd.microsoft.icon"
        href="/theme/static/favicon.ico?version=5.14.1"/>
    <link href="/theme/static/fedora-bootstrap-1.3.0/fedora-bootstrap.min.css?version=5.14.1"
        type="text/css" rel="stylesheet" />
    <link href="/theme/static/fonts/fonts.css?version=5.14.1"
        rel="stylesheet" type="text/css" />
    <link href="/theme/static/fonts/hack_fonts/css/hack-extended.min.css?version=5.14.1"
        type="text/css" rel="stylesheet" />
    <link href="/theme/static/theme.css?version=5.14.1"
        type="text/css" rel="stylesheet" />

    <link type="text/css" rel="stylesheet" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX" href="/static/vendor/font-awesome/font-awesome.css?version=5.14.1"/>
    <link type="text/css" rel="stylesheet" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX" href="/static/pagure.css?version=5.14.1"/>
<link nonce="0PleLmsbrJ9GXBoFsN8RZXLIX" rel="stylesheet" href="/static/vendor/highlight.js/styles/github.css?version=5.14.1"/>
<link nonce="0PleLmsbrJ9GXBoFsN8RZXLIX" rel="stylesheet" href="/static/vendor/highlightjs-line-numbers/highlightjs-line-numbers.min.css?version=5.14.1"/>
<style nonce="0PleLmsbrJ9GXBoFsN8RZXLIX">
  .hljs {
    background: #fff;
  }
</style>
  </head>
  <body id="home">
    
    <!-- start masthead -->
    <nav class="navbar navbar-light masthead p-0 navbar-expand">
      <div class="container">
        <a href="/" class="navbar-brand">
        <img height="40" src="/theme/static/pagure-logo.png?version=5.14.1"
             alt="pagure Logo" id="pagureLogo"/>
        </a>
        <ul class="navbar-nav ml-auto">

          

          <li class="nav-item">
            <a class="btn btn-primary" href="/login/?next=https://src.fedoraproject.org/rpms/libsmi/blob/rawhide/f/libsmi-c5830721-symbols-clash.patch">Log In</a>
          </li>
        </ul>
      </div>
    </nav>
    <!-- close masthead-->

    <div class="bodycontent">


<div class="bg-light border border-bottom pt-3">
  <div class="container">
    <div class="row mb-3">
      <div class="col-6">
        <div class="row">
        <div class="col-auto pr-0">
            <h3>
<i class="fa fa-archive text-muted"></i></h3>
        </div>
        <div class="col-auto pl-2">
            <h3 class="mb-0">
<a href="/projects/rpms/%2A">rpms</a>&nbsp;/&nbsp;<a href="/rpms/libsmi"><strong>libsmi</strong></a>
            </h3>
        </div>
        </div>
      </div>
      <div class="col-6 text-right">
        <div class="btn-group">
        <div class="btn-group">
        <a href="#"
            class="btn btn-sm dropdown-toggle btn-outline-primary"
            data-toggle="dropdown" id="watch-button">
          <i class="fa fa-clone fa-fw"></i>
          <span>Clone</span>
        </a>
        <div class="dropdown-menu dropdown-menu-right">
          <div class="m-3" id="source-dropdown">
            <div>
              <h5><strong>Source Code</strong></h5>

              <div class="form-group">
                <div class="input-group input-group-sm">
                  <div class="input-group-prepend"><span class="input-group-text">GIT</span></div>
                  <input class="form-control bg-white select-on-focus" type="text" value="https://src.fedoraproject.org/rpms/libsmi.git" readonly>
                </div>
              </div>
            </div>
          </div>
        </div>

      </div>
    </div>
  </div>
</div>

<ul class="nav nav-tabs nav-small border-bottom-0">
  <li class="nav-item mr-2 text-dark">
    <a class="nav-link active" href="/rpms/libsmi">
        <i class="fa fa-code fa-fw text-muted"></i>
        <span class="d-none d-md-inline">Source</span>
    </a>
  </li>

    <li class="nav-item mr-2 text-dark">
        <a class="nav-link" href="https://bugzilla.redhat.com/buglist.cgi?bug_status=NEW&bug_status=ASSIGNED&classification=Fedora&product=Fedora&product=Fedora EPEL&component=libsmi">
          <i class="fa fa-fw text-muted fa-exclamation-circle"></i>
          <span class="d-none d-md-inline">Issues&nbsp;</span>
          <span class="fa fa-external-link"></span>
        </a>
    </li>

    <li class="nav-item mr-2 text-dark">
      <a class="nav-link" href="/rpms/libsmi/pull-requests">
          <i class="fa fa-fw text-muted fa-arrow-circle-down"></i>
          <span class="d-none d-md-inline">Pull Requests&nbsp;</span>
          <span class="badge badge-secondary py-0 d-none d-md-inline">
            0
          </span>
      </a>
    </li>


    <li class="nav-item mr-2 text-dark">
      <a class="nav-link" href="/rpms/libsmi/stats">
          <i class="fa fa-line-chart fa-fw text-muted"></i>
          <span class="d-none d-md-inline">Stats</span>
      </a>
    </li>


</ul>
  </div>
</div>

<div class="container pt-5 repo-body-container">
  <!-- template: file.html -->
  <div class="row">
    <div class="col-2">
<nav class="nav nav-tabs nav-sidetabs flex-column flex-nowrap">
  <a class=
      "nav-link nowrap
"
      href="/rpms/libsmi">
      <i class="fa fa-home text-muted fa-fw"></i>&nbsp;<span class="d-none d-md-inline">Overview</span>
  </a>
  <a class=
    "nav-link nowrap
 active"
    href="/rpms/libsmi/tree/rawhide">
    <i class="fa fa-file-code-o text-muted fa-fw"></i>&nbsp;Files
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/libsmi/commits/rawhide">
    <i class="fa fa-list-alt text-muted fa-fw" data-glyph="spreadsheet"></i>&nbsp;Commits
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/libsmi/branches?branchname=rawhide">
    <i class="fa fa-random text-muted fa-fw"></i>&nbsp;Branches
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/libsmi/forks">
    <i class="fa fa-code-fork text-muted fa-fw"></i>&nbsp;Forks
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/libsmi/releases">
    <i class="fa fa-tags text-muted fa-fw"></i>&nbsp;Releases
  </a>

  <div class="line-height-1"></div>
  <h6>Monitoring status:</h6>
  <div class="btn-group">
    <button title="Monitoring status" class="btn btn-sm btn-outline-primary disabled"
        id="monitoring-button">
      <i id="monitoring-icon" class="fa fa-fw fa-eye"></i>
      <span id="monitoring-label" class="fa fa-circle-o-notch fa-spin fa-1x fa-fw"></span>
    </button>
  </div>

  <div class="line-height-1"></div>
  <div id="orphan-section" class="pt-3 enforce-text-break">
      <div class="line-height-1"></div>

  </div>

  <div class="pt-3">
    <div class="line-height-1">
    <h6>Bugzilla Assignee:</h6>
      <dl>
        <dt>Fedora: </dt>
        <dd id="fedora_assignee_txt">
        spot
        </dd>
        <dt>EPEL: </dt>
        <dd id="epel_assignee_txt">
        spot
        </dd>
      </dl>
    </div>
  </div>

  <div class="modal fade" id="modal_assignee" tabindex="-1"
          role="dialog" aria-labelledby="Bugzilla assignee" aria-hidden="true">
    <div class="modal-dialog" role="document">
      <div class="modal-content">
        <div class="modal-header">
          <h4 class="modal-title">Bugzilla Assignee</h4>
          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
            <span class="sr-only">Close</span>
          </button>
        </div>
        <form id="change_assignees">
          <div class="modal-body">
            <label for="fedora_assignee">Fedora</label>
            <input title="Default assignee for Fedora in bugzilla - Empty input resets to default"
              class="form-control" name="fedora_assignee" id="fedora_assignee" value="spot"/>
            <label for="epel_assignee">EPEL</label>
            <input title="Default assignee for EPEL in bugzilla (if applicable) - Empty input resets to default"
              class="form-control" name="epel_assignee" id="epel_assignee" value="spot" />
            <p class="pt-2">
              These two fields allow to specify a different default assignee for ticket opened against
              this package in bugzilla. Note: The EPEL field is
              always displayed for packages in the 'rpms' namespace regardless of whether it
              is used in bugzilla or not.            </p>
          </div>
          <div class="modal-footer">
            <button class="btn btn-secondary" type="button" title="Update bugzilla overrides" id="reset_assignees">
               Reset to defaults
            </button>
            <button class="btn btn-primary" type="submit" title="Update bugzilla overrides" id="update_assignees">
               Update
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>

  <div class="modal fade" id="modal_orphan" tabindex="-1"
          role="dialog" aria-labelledby="Orphan this package" aria-hidden="true">
    <div class="modal-dialog" role="document">
      <div class="modal-content">
        <div class="modal-header">
          <h4 class="modal-title">Orphan package</h4>
          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
            <span class="sr-only">Close</span>
          </button>
        </div>
          <div class="modal-body">
            <div class="form-group">
              <label for="reason_menu">Reason</label>
              <select id="reason_menu" class="form-control">

                <option id="lack_of_time_option_button">Lack of time</option>

                <option id="do_not_use_it_option_button">Do not use it anymore</option>

                <option id="unmaintained_option_button">Unmaintained upstream</option>

                <option id="fails_to_build_option_button">Fails to build from source</option>

                <option id="not_fixed_option_button">Important bug not fixed</option>

                <option id="other_option_button">Other</option>
              </select>
            </div>
          <label for="orphan_reason_info">Additional info</label>
          <input title="Additional info for orphaning reason"
                  class="form-control" name="orphan_reason_info" id="orphan_reason_info" value=""/>
          </div>
          <div class="modal-footer">
            <button class="btn btn-primary" type="submit" title="Orphan this package" id="orphan_button">
               Update
            </button>
          </div>
      </div>
    </div>
  </div>

  <script type="text/javascript" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX">
    window.addEventListener('load', function() {
      set_up_monitoring = function(status){
        var _label = "Disabled"
        if (status === "monitoring") {
          _label = "Monitoring";
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else if (status === "monitoring-with-scratch") {
          _label = "Scratch builds"
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else if (status === "monitoring-all") {
          _label = "Monitoring all"
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else if (status === "monitoring-all-scratch") {
          _label = "All - Scratch builds"
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else if (status === "monitoring-stable") {
          _label = "Monitoring stable"
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else if (status === "monitoring-stable-scratch") {
          _label = "Stable - Scratch builds"
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else {
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye-slash")
        }

        $("#monitoring-label").text(_label);
        $("#monitoring-label").removeClass("fa fa-circle-o-notch fa-spin fa-1x fa-fw");
      }

      $.ajax({
        url: "/_dg/anitya/rpms/libsmi",
        type: 'GET',
        dataType: 'json',
        success: function(res) {
          console.log(res);
          set_up_monitoring(res.monitoring)
        }
      });


      $("#reset_assignees").on('click', function(){
        $('#fedora_assignee').val('');
        $('#epel_assignee').val('');
        $("#change_assignees").submit();
        return false;
      });

      $("#change_assignees").on('submit',  function(){
        $('html').css('cursor', 'progress');
        $('#reset_assignees').attr('disabled', true);
        $('#update_assignees').attr('disabled', true);
        $('#update_assignees').text('Updating...');
        $.ajax({
          url: "/_dg/bzoverrides/rpms/libsmi",
          type: 'POST',
          dataType: 'json',
          data: {
             'epel_assignee': $('#epel_assignee').val(),
             'fedora_assignee': $('#fedora_assignee').val()
          },
          success: function(res) {
            $("#fedora_assignee_txt").text(res.fedora_assignee);
            $("#epel_assignee_txt").text(res.epel_assignee);
            $('#modal_assignee').modal('hide');
            $('#reset_assignees').attr('disabled', false);
            $('#update_assignees').attr('disabled', false);
            $('#update_assignees').text('Update');
            $('html').css('cursor', 'default');
            console.log("Successfully changed the bugzilla assignees");
            return false;
          },
          error: function(res) {
            var msg = '';
            if(res.responseJSON.errors){
              msg = ': ' + res.responseJSON.errors.join(', ');
            }
            alert("Unable to update the bugzilla assignee(s)" + msg);
            $('html').css('cursor', 'default');
            $('#reset_assignees').attr('disabled', false);
            $('#update_assignees').attr('disabled', false);
            $('#update_assignees').text('Update');
            return false;
          }
        })
        return false;
      });


      $.ajax({
        url: "/_dg/actived/rpms/libsmi",
        type: 'GET',
        dataType: 'json',
        success: function(res) {
          var _btn = $("#take_orphan_button");
          if (!res.active){
            _btn.off("click");
            _btn.click(function(){
                window.open(
                    "https://pagure.io/releng/new_issue?title="
                    + "Unretire rpms/libsmi"
                    + "&amp;template=package_unretirement");
            });
            _btn.prop( "title", "Package retired - Open a releng ticket to adopt it" );
            _btn.html("Retired");
          }
          _btn.removeClass('disabled');
        }
      });

      $("#orphan_button").click(function(){
        $("#orphan_button").attr("disabled", true);
        $.ajax({
          url: "/_dg/orphan/rpms/libsmi",
          type: 'POST',
          dataType: 'json',
          data: {
            'orphan_reason': $('#reason_menu').val(),
            'orphan_reason_info': $('#orphan_reason_info').val()
          },
          success: function(res) {
            $("#point_of_contact_div").html("Package is currently unmaintained");
            $("#orphan_button").attr("disabled", false);
            $('#modal_orphan').modal('hide');
            $('#orphan-section').html('');
          },
          error: function(res) {
            if (res.responseJSON.errors) {
              alert('Unable to orphan the package: ' + res.responseJSON.errors);
            } else {
              alert('Unable to orphan the package: ' + res.responseJSON.error);
            }
            $("#orphan_button").attr("disabled", false);
          }
        });
      });
    });
  </script>

</nav>    </div>
    <div class="col-10">
  <div class="row mb-1">
    <div class="col-sm-6">
    <h3>
      Files
    </h3>
    </div>

    <div class="col-sm-6 text-right">
          <div class="btn-group">
            <a href="#" class="btn btn-outline-light border-secondary text-dark btn-sm dropdown-toggle"
                    data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                    <span class="fa fa-random fa-fw"></span> Branch: <span class="font-weight-bold">rawhide</span>
            </a>
            <div class="dropdown-menu dropdown-menu-right">
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f10">
                      <span class="fa fa-random fa-fw"></span> <span class="">f10</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f11">
                      <span class="fa fa-random fa-fw"></span> <span class="">f11</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f12">
                      <span class="fa fa-random fa-fw"></span> <span class="">f12</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f13">
                      <span class="fa fa-random fa-fw"></span> <span class="">f13</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f14">
                      <span class="fa fa-random fa-fw"></span> <span class="">f14</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f15">
                      <span class="fa fa-random fa-fw"></span> <span class="">f15</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f16">
                      <span class="fa fa-random fa-fw"></span> <span class="">f16</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f17">
                      <span class="fa fa-random fa-fw"></span> <span class="">f17</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f18">
                      <span class="fa fa-random fa-fw"></span> <span class="">f18</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f19">
                      <span class="fa fa-random fa-fw"></span> <span class="">f19</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f20">
                      <span class="fa fa-random fa-fw"></span> <span class="">f20</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f21">
                      <span class="fa fa-random fa-fw"></span> <span class="">f21</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f22">
                      <span class="fa fa-random fa-fw"></span> <span class="">f22</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f23">
                      <span class="fa fa-random fa-fw"></span> <span class="">f23</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f24">
                      <span class="fa fa-random fa-fw"></span> <span class="">f24</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f25">
                      <span class="fa fa-random fa-fw"></span> <span class="">f25</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f26">
                      <span class="fa fa-random fa-fw"></span> <span class="">f26</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f27">
                      <span class="fa fa-random fa-fw"></span> <span class="">f27</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f28">
                      <span class="fa fa-random fa-fw"></span> <span class="">f28</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f29">
                      <span class="fa fa-random fa-fw"></span> <span class="">f29</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f30">
                      <span class="fa fa-random fa-fw"></span> <span class="">f30</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f31">
                      <span class="fa fa-random fa-fw"></span> <span class="">f31</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f32">
                      <span class="fa fa-random fa-fw"></span> <span class="">f32</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f33">
                      <span class="fa fa-random fa-fw"></span> <span class="">f33</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f34">
                      <span class="fa fa-random fa-fw"></span> <span class="">f34</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f35">
                      <span class="fa fa-random fa-fw"></span> <span class="">f35</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f36">
                      <span class="fa fa-random fa-fw"></span> <span class="">f36</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f37">
                      <span class="fa fa-random fa-fw"></span> <span class="">f37</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f38">
                      <span class="fa fa-random fa-fw"></span> <span class="">f38</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f39">
                      <span class="fa fa-random fa-fw"></span> <span class="">f39</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f40">
                      <span class="fa fa-random fa-fw"></span> <span class="">f40</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f41">
                      <span class="fa fa-random fa-fw"></span> <span class="">f41</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f7">
                      <span class="fa fa-random fa-fw"></span> <span class="">f7</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f8">
                      <span class="fa fa-random fa-fw"></span> <span class="">f8</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/f9">
                      <span class="fa fa-random fa-fw"></span> <span class="">f9</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/fc6">
                      <span class="fa fa-random fa-fw"></span> <span class="">fc6</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/libsmi/tree/main">
                      <span class="fa fa-random fa-fw"></span> <span class="">main</span>
                  </a>
                  <a class="dropdown-item pl-1 active" href="/rpms/libsmi/tree/rawhide">
                      <span class="fa fa-random fa-fw"></span> <span class="font-weight-bold">rawhide</span>
                  </a>
            </div>
          </div>
    </div>
  </div>
    <!-- .card -->
    <div class="card mb-3">
      <div class="card-header">
        <ol class="breadcrumb p-0 bg-transparent mb-0">
          <li class="breadcrumb-item">
            <a href="/rpms/libsmi/tree/rawhide">
              <span class="fa fa-random">
              </span>&nbsp; rawhide
            </a>
          </li>
          <li class="active breadcrumb-item">
            <span class="fa fa-file" data-glyph="">
            </span>&nbsp; libsmi-c5830721-symbols-clash.patch
          </li>
        </ol>
      </div>

  <div class="card-body p-0">
            <div class="bg-light border text-right pr-2 py-1">
                <form class="d-inline mx-2" method="POST" name="fork_project"
                    action="/fork_edit/rpms/libsmi/edit/rawhide/f/libsmi-c5830721-symbols-clash.patch">
                    <button class="btn btn-sm btn-secondary fork_project_btn">
                            Fork and Edit
                    </button>
                    
                </form>

                <a class="btn btn-secondary btn-sm" href="/rpms/libsmi/blob/rawhide/f/libsmi-c5830721-symbols-clash.patch" title="View as blob">Blob</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/libsmi/blame/libsmi-c5830721-symbols-clash.patch?identifier=rawhide" title="View git blame">Blame</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/libsmi/history/libsmi-c5830721-symbols-clash.patch?identifier=rawhide" title="View git log for this file">History</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/libsmi/raw/rawhide/f/libsmi-c5830721-symbols-clash.patch" title="View as raw">Raw</a>
            </div>

        <pre class="syntaxhighlightblock"><code class="lang-diff">diff -up libsmi-c5830721/lib/smi-data.c.clash libsmi-c5830721/lib/smi-data.c
--- libsmi-c5830721/lib/smi-data.c.clash	2021-01-15 06:10:12.000000000 -0500
+++ libsmi-c5830721/lib/smi-data.c	2024-01-05 09:22:07.729348354 -0500
@@ -78,6 +78,7 @@ static Handle   *lastHandlePtr  = NULL;
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Handle *addHandle(const char *name)
 {
     Handle *handlePtr;
@@ -112,7 +113,7 @@ Handle *addHandle(const char *name)
  *
  *----------------------------------------------------------------------
  */
-
+__attribute__ ((visibility (&#34;internal&#34;)))
 void removeHandle(Handle *handlePtr)
 {
     if (handlePtr-&gt;prevPtr) {
@@ -149,6 +150,7 @@ void removeHandle(Handle *handlePtr)
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Handle *findHandleByName(const char *name)
 {
     Handle *handlePtr;
@@ -185,7 +187,7 @@ Handle *findHandleByName(const char *nam
  *
  *----------------------------------------------------------------------
  */
-
+__attribute__ ((visibility (&#34;internal&#34;)))
 View *addView(const char *modulename)
 {
     View	      *viewPtr;
@@ -220,6 +222,7 @@ View *addView(const char *modulename)
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 int isInView(const char *modulename)
 {
     View	      *viewPtr;
@@ -257,6 +260,7 @@ int isInView(const char *modulename)
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Module *addModule(char *modulename, char *path, ModuleFlags flags,
 		  Parser *parserPtr)
 {
@@ -320,6 +324,7 @@ Module *addModule(char *modulename, char
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setModuleIdentityObject(Module *modulePtr, Object *objectPtr)
 {
     modulePtr-&gt;objectPtr = objectPtr;
@@ -343,6 +348,7 @@ void setModuleIdentityObject(Module *mod
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setModuleLastUpdated(Module *modulePtr, time_t lastUpdated)
 {
     modulePtr-&gt;lastUpdated = lastUpdated;
@@ -366,6 +372,7 @@ void setModuleLastUpdated(Module *module
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setModuleOrganization(Module *modulePtr, char *organization)
 {
     modulePtr-&gt;export.organization = organization;
@@ -392,6 +399,7 @@ void setModuleOrganization(Module *modul
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setModuleContactInfo(Module *modulePtr, char *contactinfo)
 {
     modulePtr-&gt;export.contactinfo = contactinfo;
@@ -415,6 +423,7 @@ void setModuleContactInfo(Module *module
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setModuleDescription(Module *modulePtr, char *description,
 			  Parser *parserPtr)
 {
@@ -446,6 +455,7 @@ void setModuleDescription(Module *module
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setModuleReference(Module *modulePtr, char *reference, Parser *parserPtr)
 {
     if (modulePtr-&gt;export.reference) {
@@ -478,6 +488,7 @@ void setModuleReference(Module *modulePt
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Module *findModuleByName(const char *modulename)
 {
     Module	*modulePtr;
@@ -511,6 +522,7 @@ Module *findModuleByName(const char *mod
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Revision *addRevision(time_t date, char *description, Parser *parserPtr)
 {
     Revision	  *revisionPtr, *r;
@@ -575,6 +587,7 @@ Revision *addRevision(time_t date, char
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setRevisionLine(Revision *revisionPtr, int line, Parser *parserPtr)
 {
     if (line) {
@@ -604,6 +617,7 @@ void setRevisionLine(Revision *revisionP
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Import *addImport(char *name, Parser *parserPtr)
 {
     Import        *importPtr;
@@ -650,6 +664,7 @@ Import *addImport(char *name, Parser *pa
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void addImportFlags(Import *importPtr, ImportFlags flags)
 {
     importPtr-&gt;flags |= flags;
@@ -673,6 +688,7 @@ void addImportFlags(Import *importPtr, I
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setImportModulename(Import *importPtr, char *modulename)
 {
     if (importPtr-&gt;export.module) {
@@ -701,6 +717,7 @@ void setImportModulename(Import *importP
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 int checkImports(Module *modulePtr, Parser *parserPtr)
 {
     int         n = 0;
@@ -768,6 +785,7 @@ int checkImports(Module *modulePtr, Pars
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Import *findImportByName(const char *name, Module *modulePtr)
 {
     Import           *importPtr;
@@ -807,6 +825,7 @@ Import *findImportByName(const char *nam
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Import *findImportByModulenameAndName(const char *modulename,
 				      const char *name, Module *modulePtr)
 {
@@ -844,6 +863,7 @@ Import *findImportByModulenameAndName(co
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Object *addObject(char *objectname, Node *parentNodePtr, SmiSubid subid,
 		  ObjectFlags flags, Parser *parserPtr)
 {
@@ -941,6 +961,7 @@ Object *addObject(char *objectname, Node
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Object *duplicateObject(Object *templatePtr, ObjectFlags flags,
 			Parser *parserPtr)
 {
@@ -1021,6 +1042,7 @@ Object *duplicateObject(Object *template
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Node *addNode (Node *parentNodePtr, SmiSubid subid, NodeFlags flags,
 	       Parser *parserPtr)
 {
@@ -1098,6 +1120,7 @@ Node *addNode (Node *parentNodePtr, SmiS
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Node *createNodes(unsigned int oidlen, SmiSubid *oid)
 {
     Node	 *parentNodePtr, *nodePtr;
@@ -1135,6 +1158,7 @@ Node *createNodes(unsigned int oidlen, S
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Node *createNodesByOidString(const char *oid)
 {
     char		*p, *elements;
@@ -1177,6 +1201,7 @@ Node *createNodesByOidString(const char
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Node *getParentNode(Node *nodePtr)
 {
     return nodePtr-&gt;parentPtr;
@@ -1201,6 +1226,7 @@ Node *getParentNode(Node *nodePtr)
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 static void mergeNodeTrees(Node *toNodePtr, Node *fromNodePtr,
 			   Parser *parserPtr)
 {
@@ -1301,6 +1327,7 @@ static void mergeNodeTrees(Node *toNodeP
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Object *setObjectName(Object *objectPtr, char *name, Parser *parserPtr)
 {
     Node	      *nodePtr, *nextPtr;
@@ -1401,6 +1428,7 @@ Object *setObjectName(Object *objectPtr,
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectType(Object *objectPtr, Type *typePtr)
 {
     objectPtr-&gt;typePtr = typePtr;
@@ -1424,6 +1452,7 @@ void setObjectType(Object *objectPtr, Ty
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectAccess(Object *objectPtr, SmiAccess access)
 {
     objectPtr-&gt;export.access = access;
@@ -1447,6 +1476,7 @@ void setObjectAccess(Object *objectPtr,
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectStatus(Object *objectPtr, SmiStatus status)
 {
     objectPtr-&gt;export.status = status;
@@ -1470,6 +1500,7 @@ void setObjectStatus(Object *objectPtr,
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectDescription(Object *objectPtr, char *description,
 			  Parser *parserPtr)
 {
@@ -1501,6 +1532,7 @@ void setObjectDescription(Object *object
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectReference(Object *objectPtr, char *reference, Parser *parserPtr)
 {
     if (objectPtr-&gt;export.reference) {
@@ -1531,6 +1563,7 @@ void setObjectReference(Object *objectPt
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectFormat(Object *objectPtr, char *format)
 {
     if (objectPtr-&gt;export.format) smiFree(objectPtr-&gt;export.format);
@@ -1555,6 +1588,7 @@ void setObjectFormat(Object *objectPtr,
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectUnits(Object *objectPtr, char *units)
 {
     if (objectPtr-&gt;export.units) smiFree(objectPtr-&gt;export.units);
@@ -1579,6 +1613,7 @@ void setObjectUnits(Object *objectPtr, c
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectDecl(Object *objectPtr, SmiDecl decl)
 {
     objectPtr-&gt;export.decl = decl;
@@ -1602,6 +1637,7 @@ void setObjectDecl(Object *objectPtr, Sm
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectLine(Object *objectPtr, int line, Parser *parserPtr)
 {
     if (line) {
@@ -1629,6 +1665,7 @@ void setObjectLine(Object *objectPtr, in
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectNodekind(Object *objectPtr, SmiNodekind nodekind)
 {
     objectPtr-&gt;export.nodekind = nodekind;
@@ -1652,6 +1689,7 @@ void setObjectNodekind(Object *objectPtr
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void addObjectFlags(Object *objectPtr, ObjectFlags flags)
 {
     objectPtr-&gt;flags |= flags;
@@ -1675,6 +1713,7 @@ void addObjectFlags(Object *objectPtr, O
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void deleteObjectFlags(Object *objectPtr, ObjectFlags flags)
 {
     objectPtr-&gt;flags &amp;= ~flags;
@@ -1698,6 +1737,7 @@ void deleteObjectFlags(Object *objectPtr
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 int checkObjectFlags(Object *objectPtr, ObjectFlags flags)
 {
     return ((objectPtr-&gt;flags &amp; flags) == flags);
@@ -1748,6 +1788,7 @@ void setObjectIndex(Object *objectPtr, I
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectList(Object *objectPtr, List *listPtr)
 {
     objectPtr-&gt;listPtr = listPtr;
@@ -1771,6 +1812,7 @@ void setObjectList(Object *objectPtr, Li
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectRelated(Object *objectPtr, Object *relatedPtr)
 {
     objectPtr-&gt;relatedPtr = relatedPtr;
@@ -1794,6 +1836,7 @@ void setObjectRelated(Object *objectPtr,
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectImplied(Object *objectPtr, int implied)
 {
     objectPtr-&gt;export.implied = implied;
@@ -1817,6 +1860,7 @@ void setObjectImplied(Object *objectPtr,
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectCreate(Object *objectPtr, int create)
 {
     objectPtr-&gt;export.create = create;
@@ -1840,6 +1884,7 @@ void setObjectCreate(Object *objectPtr,
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectIndexkind(Object *objectPtr, SmiIndexkind indexkind)
 {
     objectPtr-&gt;export.indexkind = indexkind;
@@ -1863,6 +1908,7 @@ void setObjectIndexkind(Object *objectPt
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectValue(Object *objectPtr, SmiValue *valuePtr)
 {
     objectPtr-&gt;export.value = *valuePtr;
@@ -1887,6 +1933,7 @@ void setObjectValue(Object *objectPtr, S
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setObjectUniqueness(Object *objectPtr, List *listPtr)
 {
     objectPtr-&gt;uniquenessPtr = listPtr;
@@ -1934,6 +1981,7 @@ void setObjectUniqueness(Object *objectP
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Node *findNodeByParentAndSubid(Node *parentNodePtr, SmiSubid subid)
 {
     Node *nodePtr;
@@ -1970,6 +2018,7 @@ Node *findNodeByParentAndSubid(Node *par
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Node *findNodeByOid(unsigned int oidlen, SmiSubid *oid)
 {
     Node          *nodePtr;
@@ -2002,6 +2051,7 @@ Node *findNodeByOid(unsigned int oidlen,
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Node *findNodeByOidString(char *oid)
 {
     Node *nodePtr;
@@ -2039,6 +2089,7 @@ Node *findNodeByOidString(char *oid)
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Object *findObjectByNode(Node *nodePtr)
 {
     Object    *objectPtr;
@@ -2081,6 +2132,7 @@ Object *findObjectByNode(Node *nodePtr)
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Object *findObjectByModuleAndNode(Module *modulePtr, Node *nodePtr)
 {
     Object    *objectPtr;
@@ -2116,6 +2168,7 @@ Object *findObjectByModuleAndNode(Module
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Object *findObjectByModulenameAndNode(const char *modulename, Node *nodePtr)
 {
     Object     *objectPtr;
@@ -2151,6 +2204,7 @@ Object *findObjectByModulenameAndNode(co
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Object *findObjectByName(const char *objectname)
 {
     Module	     *modulePtr;
@@ -2195,6 +2249,7 @@ Object *findObjectByName(const char *obj
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Object *findNextObjectByName(const char *objectname, Object *prevObjectPtr)
 {
     Module	     *modulePtr;
@@ -2238,6 +2293,7 @@ Object *findNextObjectByName(const char
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Object *findObjectByModulenameAndName(const char *modulename,
 				      const char *objectname)
 {
@@ -2291,6 +2347,7 @@ Object *findObjectByModulenameAndName(co
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Object *findObjectByModuleAndName(Module *modulePtr, const char *objectname)
 {
     Object	  *objectPtr;
@@ -2340,6 +2397,7 @@ Object *findObjectByModuleAndName(Module
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Type *addType(char *type_name, SmiBasetype basetype, TypeFlags flags,
 	      Parser *parserPtr)
 {
@@ -2401,6 +2459,7 @@ Type *addType(char *type_name, SmiBasety
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Type *duplicateType(Type *templatePtr, TypeFlags flags, Parser *parserPtr)
 {
     Type		  *typePtr;
@@ -2457,6 +2516,7 @@ Type *duplicateType(Type *templatePtr, T
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Type *setTypeName(Type *typePtr, char *name)
 {
     Type              *type2Ptr;
@@ -2549,6 +2609,7 @@ Type *setTypeName(Type *typePtr, char *n
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setTypeParent(Type *typePtr, Type *parentPtr)
 {
     typePtr-&gt;parentPtr = parentPtr;
@@ -2572,6 +2633,7 @@ void setTypeParent(Type *typePtr, Type *
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setTypeStatus(Type *typePtr, SmiStatus status)
 {
     typePtr-&gt;export.status = status;
@@ -2595,6 +2657,7 @@ void setTypeStatus(Type *typePtr, SmiSta
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setTypeBasetype(Type *typePtr, SmiBasetype basetype)
 {
     typePtr-&gt;export.basetype = basetype;
@@ -2618,6 +2681,7 @@ void setTypeBasetype(Type *typePtr, SmiB
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setTypeDescription(Type *typePtr, char *description, Parser *parserPtr)
 {
     if (typePtr-&gt;export.description) {
@@ -2649,6 +2713,7 @@ void setTypeDescription(Type *typePtr, c
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setTypeReference(Type *typePtr, char *reference, Parser *parserPtr)
 {
     if (typePtr-&gt;export.reference) {
@@ -2684,6 +2749,7 @@ void setTypeReference(Type *typePtr, cha
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setTypeList(Type *typePtr, List *listPtr)
 {
     if (!typePtr-&gt;listPtr) {
@@ -2709,6 +2775,7 @@ void setTypeList(Type *typePtr, List *li
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setTypeFormat(Type *typePtr, char *format)
 {
     if (typePtr-&gt;export.format) smiFree(typePtr-&gt;export.format);
@@ -2734,6 +2801,7 @@ void setTypeFormat(Type *typePtr, char *
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setTypeUnits(Type *typePtr, char *units)
 {
     if (typePtr-&gt;export.units) smiFree(typePtr-&gt;export.units);
@@ -2758,6 +2826,7 @@ void setTypeUnits(Type *typePtr, char *u
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setTypeDecl(Type *typePtr, SmiDecl decl)
 {
     typePtr-&gt;export.decl = decl;
@@ -2781,6 +2850,7 @@ void setTypeDecl(Type *typePtr, SmiDecl
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setTypeLine(Type *typePtr, int line, Parser *parserPtr)
 {
     if (line) {
@@ -2808,6 +2878,7 @@ void setTypeLine(Type *typePtr, int line
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setTypeValue(Type *typePtr, SmiValue *valuePtr)
 {
     typePtr-&gt;export.value = *valuePtr;
@@ -2831,6 +2902,7 @@ void setTypeValue(Type *typePtr, SmiValu
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void addTypeFlags(Type *typePtr, TypeFlags flags)
 {
     typePtr-&gt;flags |= flags;
@@ -2854,6 +2926,7 @@ void addTypeFlags(Type *typePtr, TypeFla
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void deleteTypeFlags(Type *typePtr, TypeFlags flags)
 {
     typePtr-&gt;flags &amp;= ~flags;
@@ -2878,6 +2951,7 @@ void deleteTypeFlags(Type *typePtr, Type
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Type * findTypeByName(const char *type_name)
 {
     Module *modulePtr;
@@ -2916,6 +2990,7 @@ Type * findTypeByName(const char *type_n
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Type *findNextTypeByName(const char *type_name, Type *prevTypePtr)
 {
     Module *modulePtr;
@@ -2954,6 +3029,7 @@ Type *findNextTypeByName(const char *typ
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Type *findTypeByModulenameAndName(const char *modulename,
 				  const char *type_name)
 {
@@ -2993,6 +3069,7 @@ Type *findTypeByModulenameAndName(const
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Type *findTypeByModuleAndName(Module *modulePtr, const char *type_name)
 {
     Type        *typePtr;
@@ -3025,6 +3102,7 @@ Type *findTypeByModuleAndName(Module *mo
  *
  *---------------------------------------------------------------------- */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 NamedNumber *findTypeNamedNumber(Type *typePtr,
 				 SmiInteger32 number)
 {
@@ -3058,6 +3136,7 @@ NamedNumber *findTypeNamedNumber(Type *t
  *
  *---------------------------------------------------------------------- */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Identity *addIdentity(char *identityname, Parser *parserPtr)
 {
 	Identity  *identityPtr;
@@ -3104,6 +3183,7 @@ Identity *addIdentity(char *identityname
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setIdentityDecl(Identity *identityPtr, SmiDecl decl)
 {
     identityPtr-&gt;export.decl = decl;
@@ -3125,6 +3205,7 @@ void setIdentityDecl(Identity *identityP
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setIdentityStatus(Identity *identityPtr, SmiStatus status)
 {
     identityPtr-&gt;export.status = status;
@@ -3148,6 +3229,7 @@ void setIdentityStatus(Identity *identit
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setIdentityDescription(Identity *identityPtr, char *description, Parser *parserPtr)
 {
     if (identityPtr-&gt;export.description) {
@@ -3179,6 +3261,7 @@ void setIdentityDescription(Identity *id
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setIdentityReference(Identity *identityPtr, char *reference, Parser *parserPtr)
 {
     if (identityPtr-&gt;export.reference) {
@@ -3208,6 +3291,7 @@ void setIdentityReference(Identity *iden
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setIdentityParent(Identity *identityPtr, Identity *parentPtr)
 {
     if(identityPtr) identityPtr-&gt;parentPtr = parentPtr;
@@ -3231,6 +3315,7 @@ void setIdentityParent(Identity *identit
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Identity *findIdentityByName(const char *identityname)
 {
     Module *modulePtr;
@@ -3267,6 +3352,7 @@ Identity *findIdentityByName(const char
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Identity *findIdentityByModuleAndName(Module *modulePtr,
 									const char *identityname)
 {
@@ -3302,6 +3388,7 @@ Identity *findIdentityByModuleAndName(Mo
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Identity *findIdentityByModulenameAndName(const char *modulename,
 				  const char *identity_name)
 {
@@ -3340,6 +3427,7 @@ Identity *findIdentityByModulenameAndNam
  *
  *---------------------------------------------------------------------- */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Class *addClass(char *classname, Parser *parserPtr)
 {
 	Class  *classPtr;
@@ -3391,6 +3479,7 @@ Class *addClass(char *classname, Parser
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setClassDecl(Class *classPtr, SmiDecl decl)
 {
     classPtr-&gt;export.decl = decl;
@@ -3412,6 +3501,7 @@ void setClassDecl(Class *classPtr, SmiDe
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setClassStatus(Class *classPtr, SmiStatus status)
 {
     classPtr-&gt;export.status = status;
@@ -3435,6 +3525,7 @@ void setClassStatus(Class *classPtr, Smi
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setClassDescription(Class *classPtr, char *description, Parser *parserPtr)
 {
     if (classPtr-&gt;export.description) {
@@ -3466,6 +3557,7 @@ void setClassDescription(Class *classPtr
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setClassReference(Class *classPtr, char *reference, Parser *parserPtr)
 {
     if (classPtr-&gt;export.reference) {
@@ -3495,6 +3587,7 @@ void setClassReference(Class *classPtr,
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setClassParent(Class *classPtr, Class *parentPtr)
 {
     if(classPtr) classPtr-&gt;parentPtr = parentPtr;
@@ -3516,6 +3609,7 @@ void setClassParent(Class *classPtr, Cla
  *
  *----------------------------------------------------------------------
  */
+__attribute__ ((visibility (&#34;internal&#34;)))
 Class *findClassByModuleAndName(Module *modulePtr,char *name)
 {
     Class   *classPtr;
@@ -3548,6 +3642,7 @@ Class *findClassByModuleAndName(Module *
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Class *findClassByModulenameAndName(const char *modulename,
 				  const char *class_name)
 {
@@ -3586,6 +3681,7 @@ Class *findClassByModulenameAndName(cons
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Attribute *duplicateTypeToAttribute(Type *templatePtr, Class *classPtr, Parser *parserPtr)
 {
     Attribute		  *attributePtr;
@@ -3640,6 +3736,7 @@ Attribute *duplicateTypeToAttribute(Type
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Attribute *addAttribute(char *attribute_name,
 	      Class *classPtr, Parser *parserPtr)
 {
@@ -3694,6 +3791,7 @@ Attribute *addAttribute(char *attribute_
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setAttributeDecl(Attribute *attributePtr, SmiDecl decl)
 {
     attributePtr-&gt;export.decl = decl;
@@ -3715,6 +3813,7 @@ void setAttributeDecl(Attribute *attribu
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setAttributeParentType(Attribute *attributePtr, Type *parentPtr)
 {
     attributePtr-&gt;parentTypePtr = parentPtr;
@@ -3736,6 +3835,7 @@ void setAttributeParentType(Attribute *a
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setAttributeParentClass(Attribute *attributePtr, Class *parentPtr)
 {
     attributePtr-&gt;parentClassPtr = parentPtr;
@@ -3760,6 +3860,7 @@ void setAttributeParentClass(Attribute *
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setAttributeList(Attribute *attributePtr, List *listPtr)
 {
     if (!attributePtr-&gt;listPtr) {
@@ -3783,6 +3884,7 @@ void setAttributeList(Attribute *attribu
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setAttributeName(Attribute *attributePtr, char *name)
 {
     
@@ -3804,6 +3906,7 @@ void setAttributeName(Attribute *attribu
  *
  *----------------------------------------------------------------------
  */
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setAttributeAccess(Attribute *attributePtr,SmiAccess access)
 {
 	attributePtr-&gt;export.access = access;
@@ -3826,6 +3929,7 @@ void setAttributeAccess(Attribute *attri
  *
  *----------------------------------------------------------------------
  */
+__attribute__ ((visibility (&#34;internal&#34;)))
 Event *addEvent(char *eventname, Class *classPtr, 
 		             Parser *parserPtr)
 {
@@ -3875,6 +3979,7 @@ Event *addEvent(char *eventname, Class *
  *
  *---------------------------------------------------------------------- */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Macro *addMacro(char *macroname, MacroFlags flags, Parser *parserPtr)
 {
     Macro	  *macroPtr;
@@ -3924,6 +4029,7 @@ Macro *addMacro(char *macroname, MacroFl
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setMacroStatus(Macro *macroPtr, SmiStatus status)
 {
     macroPtr-&gt;export.status = status;
@@ -3947,6 +4053,7 @@ void setMacroStatus(Macro *macroPtr, Smi
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setMacroDescription(Macro *macroPtr, char *description, Parser *parserPtr)
 {
     if (macroPtr-&gt;export.description) {
@@ -3978,6 +4085,7 @@ void setMacroDescription(Macro *macroPtr
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setMacroReference(Macro *macroPtr, char *reference, Parser *parserPtr)
 {
     if (macroPtr-&gt;export.reference) {
@@ -4007,6 +4115,7 @@ void setMacroReference(Macro *macroPtr,
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setMacroAbnf(Macro *macroPtr, char *abnf, Parser *parserPtr)
 {
     if (macroPtr-&gt;export.abnf) {
@@ -4038,6 +4147,7 @@ void setMacroAbnf(Macro *macroPtr, char
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setMacroDecl(Macro *macroPtr, SmiDecl decl)
 {
     macroPtr-&gt;export.decl = decl;
@@ -4061,6 +4171,7 @@ void setMacroDecl(Macro *macroPtr, SmiDe
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 void setMacroLine(Macro *macroPtr, int line, Parser *parserPtr)
 {
     if (line) {
@@ -4089,6 +4200,7 @@ void setMacroLine(Macro *macroPtr, int l
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Macro *findMacroByName(const char *macroname)
 {
     Module *modulePtr;
@@ -4127,6 +4239,7 @@ Macro *findMacroByName(const char *macro
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Macro *findMacroByModuleAndName(Module *modulePtr, const char *macroname)
 {
     Macro      *macroPtr;
@@ -4162,6 +4275,7 @@ Macro *findMacroByModuleAndName(Module *
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 NamedNumber *findNamedNumberByName(Type *typePtr,const char *name)
 {
 	List *listPtr;
@@ -4593,6 +4707,7 @@ void smiFreeData()
  *----------------------------------------------------------------------
  */
 
+__attribute__ ((visibility (&#34;internal&#34;)))
 Module *loadModule(const char *modulename, Parser *parserPtr)
 {
     Parser	    parser;
</code></pre>
  </div>
 </div> <!-- end .card-->

</div>
</div>
<!-- /template: file.html -->
</div>
    </div>

        <div class="footer pt-4 text-white">
        <div class="container">
            <div class="d-flex align-items-center">
                <div>
                    <div>Powered by <a href="https://pagure.io/pagure" class="notblue">Pagure</a> 5.14.1</div>
                    <div>
                        <a href="https://docs.pagure.org/pagure/usage/index.html" class="notblue">Documentation</a> &bull;
                        <a href="https://pagure.io/pagure/new_issue" class="notblue">File an Issue</a> &bull;
                        <a href="/about">About this Instance</a> &bull;
                        <a href="/ssh_info" class="notblue">SSH Hostkey/Fingerprint</a>
                    </div>
                </div>
                <div class="ml-auto text-right">
                    <div>&copy; Red Hat, Inc. and others.</div>
                </div>
            </div>
        </div>
    </div>


    <script type="text/javascript" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX" src="/static/vendor/jquery/jquery.min.js?version=5.14.1"></script>

    <script src="/static/vendor/bootstrap/bootstrap.bundle.min.js?version=5.14.1"></script>

    <script type="text/javascript" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX">
      $('[data-toggle="tooltip"]').tooltip({placement : 'bottom'});
      $(".cancel_btn").click(function() {
        history.back();
      });
    </script>

<script type="text/javascript"  nonce="0PleLmsbrJ9GXBoFsN8RZXLIX" src="/static/vendor/lazyload/lazyload.min.js?version=5.14.1"></script>

<script type="text/javascript" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX">
window.addEventListener("load", function(event) {
    lazyload();
});
</script>

<script type="text/javascript" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX">
$("#giturl-toggle").on('click', function(event){
  event.stopPropagation();
  $("#giturl-more").toggle();
  $("#giturl-toggle").hide();
})

$(".fork_project_btn").click(function() {
  $('#fork_project').submit();
});

$(".select-on-focus").on("focus", function() {
  $(this).select();
});

</script>

<script type="text/javascript" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX" src="/static/vendor/highlight.js/highlight.pack.js?version=5.14.1"></script>
<script type="text/javascript" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX" src="/static/vendor/highlightjs-line-numbers/highlightjs-line-numbers.min.js?version=5.14.1"></script>
<script type="text/javascript" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX" src="/static/vendor/highlight.js/spec.js?version=5.14.1"></script>

<script type="text/javascript" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX">
  $(document).ready(function() {
  $('.fork_project_btn').click($("[name=fork_project]").submit);

  $('pre.syntaxhighlightblock code').each(function(i, block) {
    hljs.highlightBlock(block);
    hljs.lineNumbersBlock(block);
  });

  var cls = "highlighted-line";
  var lines = location.hash.substr(2).split('-').map(function (x) { return parseInt(x, 10) });
  if (! isNaN(lines[0]))
  {
    for (var i = lines[lines.length - 1]; i >= lines[0]; i--) {
      $('#_' + i).parent().parent().addClass(cls);
    }
    setTimeout(function(){
      $("#_" + lines[0]).get(0).scrollIntoView({behavior: "instant", block: "start", inline: "nearest"});
    }, 50);
  }
});
</script>

<script type="text/javascript" nonce="0PleLmsbrJ9GXBoFsN8RZXLIX">

  function updateHighlight() {
    var cls = "highlighted-line";
    $('.' + cls).removeClass(cls)
    if (location.hash !== '') {
      var lines = location.hash.substr(2).split('-').map(function (x) { return parseInt(x, 10) });
      for (var i = lines[lines.length - 1]; i >= lines[0]; i--) {
        $('[data-line-number=' + i + ']').closest('tr').addClass(cls);
      }
      return lines;
    }
    return [];
  }
  $(window).on('hashchange', updateHighlight);
  var selected = [];
  $("[data-line-number]").click(function (ev) {
    var line = $(this).attr('data-line-number');
    if (ev.shiftKey) {
      selected = selected.slice(-1).concat(line);
    } else {
      selected = [line];
    }

    var hash = '_' + selected[0];
    if (selected.length === 2) {
      hash = '_' + Math.min(selected[0], selected[1]) + '-' + Math.max(selected[0], selected[1]);
    }
    window.location.hash = hash;
    return false;
  });

</script>


</body>
</html>